components/AppBar/src/MDCAppBar.m

```diff
@@ -32,6 +33,37 @@
 static NSString *const MDCAppBarHeaderStackViewKey = @"MDCAppBarHeaderStackViewKey";
 static const CGFloat kStatusBarHeight = 20;
 
+@implementation MDCAppBarTextColorAccessibilityMutator
+
+- (nonnull instancetype)init {
+  self = [super init];
+  return self;
+}
+
+- (void)mutate:(nonnull MDCAppBar *)appBar {
+  // Determine what is the appropriate background color
+  // Because navigation bar renders above headerview, it takes presedence
+  UIColor *backgroundColor = appBar.navigationBar.backgroundColor ?:
+      appBar.headerViewController.headerView.backgroundColor;
+  if (!backgroundColor) {
+    return;
+  }
+
+  // Update title label color based on navigationBar/headerView backgroundColor
+  NSMutableDictionary *textAttr =
+      [NSMutableDictionary dictionaryWithDictionary:[appBar.navigationBar titleTextAttributes]];
+  UIColor *textColor = [MDFTextAccessibility textColorOnBackgroundColor:backgroundColor
```

Comment:
we now need to use the Typography's [isLargeForContrastRatios](https://github.com/material-components/material-components-ios/blob/develop/components/Typography/src/MDCTypography.h#L163) 

since some fonts want medium to be large